xdk/build.gradle.kts

```diff
@@ -264,7 +265,7 @@ val compileJsonDB = tasks.register<JavaExec>("compileJsonDB") {
     jvmArgs("-Xms1024m", "-Xmx1024m", "-ea")
 
     classpath(javatoolsJar)
-    args("-o", "$libDir",
+    args("-o", "$libDir", "$seed",
          "-version", "$xdkVersion",
```

Comment:
Except for actually adding several changes for the master build script for XTC, making it possible to run it with a seed - currently something you HARD CODE in the build logic is bad.

1) This is the master build. Its concern is not to be testable
2) Should you still require it to be testable, then use the test configuration . 
3) The seed is a hard coded variable, which means that if we change it, we add a dependency that basically triggers a manual full rebuild and gradle having to update graphs, and redo whatever isn't cached. That's not OK. If I want to pass random seeds to gradle (which is doubtful), that is done through a property -Pseed on the command line, or as an environment variable or as a java SystemProperty or something. Gradle then takes that value and uses it as input the compiler task, or has the compiler task just read it using some standard environment mechanism. 
4) -seed as a first class citizen of compiler arguments, placing it in the same category as -o, -L, -version and other compiler flags that actually have to do with the compilation. Compiler options are documented and public. Documenting that seed exists and publicising it not what we want to do. 
5) If we want to run the compiler on anything, at the moment, in the test suite, so as not to have to deal with gradle, that works fine. Add a unit test class under javatools/src/test/org/vm/compiler/RandomListFilesTest, and have it call compile jobs there... e.g. new Compiler(...arguments..).process() or something. Do not involve anything where a test state causes a project rebuild. There is no end to the trouble that will cause, including unnecessary rebuilds.

Reply:
All are valid and in my view warrant a discussion. I was thinking about #4 quite a bit myself, but in terms of providing some more generic "global inject-like" facility (kind of Java's -D flag). All my previous attempts were stricken down by Cam's critics though )

Reply:
Yeah, well I got the conversation rolling anyways.
- "Write in the local style", totally agree, will change.
- No IDEA integration for me (yet).  Emacs does not grok the Tangosol style, and fights me towards a more normal Java format.  So no "Ctrl-Alt-L and it sorts itself out".
- Using the JDK shuffle: Sounds good to me.  Didn't know it was there.
- "Not random enough".  Disagree; in my experience it doesn't take very much random to trigger these things.  But the JDK shuffle should do a good job in any case.
- Seed logic in the master gradle: Dunno where else to put it, so feedback welcome.
- "NO visibility of seed to other parts".  That wasn't a goal of this PR.  This PR was simply to (1) test the PR process, and (2) more complete file-order shuffle to see if there was another lurker, and (3) start the process (but not complete it) for better testing.

Markus's long comment gets a long reply in another comment.

Reply:
Markus: 
(1),(2) - keep master build clean of test state: 100% agree
(3) - First class argument citizen "-seed" is bad, 100% agree.  But I don't see a secret "-XXseed=123" parser either.  Not sure we need a secret arg parser.  But still we need a way to inject a seed for testing... which falls under (4a) below.
(5) "There are several listFiles... Extending this mechanism...".  Nah, slippery slope argument, does not apply.  You don't want to extend this mechanism to all things, and neither do I.  This is a specific targeted place where we hit a bug in the past.  If you like, consider it a targeted regression test that happens to use a very flexible mechanism we might want to use in other places in the future.  
(4a) "Instead, add a unit test to ...", first time I got somebody to admit we had a place for unit tests!  This is how I do it for my AA project as well.

(4b) "is supposed to be done" ... 
There's NO "supposed to be done" without a team agreement, and at the moment, I do not agree to the use of mock's where a trivial line of code does the job.  We can have that team-wide discussion and come to a mutual agreement, but we have not yet do so.